hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/EntityNum.java

```diff
@@ -77,11 +75,6 @@ public static EntityNum fromAccountId(final AccountID grpc) {
 
     public static EntityNum fromEvmAddress(final Address address) {
         final var bytes = address.toArrayUnsafe();
-        final var shard = shardFromEvmAddress(bytes);
```

Comment:
🤔 Was looking closer at this one, pondering if it is a safe change, since before this PR it meant that any non-mirror address passed to [`StaticEntityAccess.getBalance()`](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/StaticEntityAccess.java#L115) would always throw a NPE: 
```
    public long getBalance(final Address address) {
        return accounts.get(fromEvmAddress(address)).getBalance();
    }
```

And I realized actually we _do_ only pass mirror addresses into `StateEntityAccess.getBalance()`, because it's only called [here](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/HederaEvmWorldState.java#L69); and the `address` in that call site has already been unaliased by the subclass `get()` implementation [here](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/AbstractLedgerWorldUpdater.java#L146).

Which means afaict `HederaEvmWorldState.get()` **does** only return `Account` objects with mirror addresses from [here](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/HederaEvmWorldState.java#L70). (I observed this once, but then convinced myself otherwise. 😁)

Thus if Besu calls `WorldUpdater.get(Address)` with a non-mirror address, the `Account` object we return will instead have the mirror version of the address. I find it hard to ponder the implications here... @stoqnkpL, @dimitar-dinev, @IvanKavaldzhiev, any thoughts?)

Reply:
Hey @tinker-michaelj , I think an important point is that we remove HederaEvmWorldState and 2 other classes (we will copy them in mirror-node) in this [PR](https://github.com/hashgraph/hedera-services/pull/5112), so I believe we shouldn't take into account any implications for them in reviewing this PR.

Reply:
Ah got it @IvanKavaldzhiev, will take my puzzlement over to that PR. 😂